src/TServer.cpp

```diff
@@ -15,6 +15,67 @@
 
 #include "Json.h"
 
+static std::optional<std::pair<int, int>> GetPidVid(std::string str) {
```

Comment:
this is performance-critical, so we want to avoid a copy here and pass by (const) ref instead
```suggestion
static std::optional<std::pair<int, int>> GetPidVid(const std::string& str) {
```